### DISCUSSION TITLE
dynamic apostrophe-login "email.from" address / domain

### DISCUSSION THREAD

**Discussion by paulgrieselhuber** [apostrophecms/apostrophe#3403] [Q&A]
When extending `apostrophe-login` in a multisite configuration, is it possible to have a dynamic `from` address?

For example, we are currently doing:

```
module.exports = {
  email: {
    from: "password-reset@website.com"
  }
}
```

But I would like to be able to do something like:

```
module.exports = {
  email: {
    from: (site) => site.supportEmail
  }
}
```

I know the syntax will differ, but that concept is the objective. Is it possible to pass the "site" (or something equally referential) to this module here?

**Comment by boutell** [comment]
Your `sites/index.js` can export a function that takes the `site` object
and returns an app.js-style config object, rather than just exporting a
config object directly.

Using that, you can pass an option powered by a property of `site` to any
module you like.

On Thu, Sep 9, 2021 at 1:02 PM Paul Grieselhuber ***@***.***>
wrote:

> When extending apostrophe-login in a multisite configuration, is it
> possible to have a dynamic from address?
>
> For example, we are currently doing:
>
> module.exports = {
>   email: {
>     from: ***@***.***"
>   }
> }
>
> But I would like to be able to do something like:
>
> module.exports = {
>   email: {
>     from: (site) => site.supportEmail
>   }
> }
>
> I know the syntax will differ, but that concept is the objective. Is it
> possible to pass the "site" (or something equally referential) to this
> module here?
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/apostrophecms/apostrophe/discussions/3403>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAAH27LDJFGKU5DXMJIF7FDUBDSDXANCNFSM5DXSMVTA>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
>


-- 

THOMAS BOUTELL | CHIEF TECHNOLOGY OFFICER
APOSTROPHECMS | apostrophecms.com | he/him/his
